### PR TITLE
Macros with arguments

### PR DESCRIPTION
**Pull request by dlubarov**
See `basic_macros.rs` for an example. Will also let us write macros like `%mul_by(2)` which I think will be handy.

cc @nbgl

### CODE REVIEW

File: evm/src/cpu/kernel/assembler.rs [hunk 1]
```diff
@@ -41,33 +55,31 @@ pub(crate) fn assemble(files: Vec<File>) -> Kernel {
...
 
-fn find_macros(files: &[File]) -> HashMap<String, Vec<Item>> {
+fn find_macros(files: &[File]) -> HashMap<String, Macro> {
     let mut macros = HashMap::new();
     for file in files {
         for item in &file.body {
-            if let Item::MacroDef(name, items) = item {
-                macros.insert(name.clone(), items.clone());
+            if let Item::MacroDef(name, params, items) = item {
+                let _macro = Macro {
```
Comment (wborgeaud) [inline evm/src/cpu/kernel/assembler.rs:63]: nit: Why the underscore?

File: evm/src/cpu/kernel/ast.rs [hunk 2]
```diff
@@ -29,6 +29,7 @@ pub(crate) enum Item {
 pub(crate) enum PushTarget {
     Literal(Literal),
     Label(String),
+    Var(String),
```
Comment (wborgeaud) [inline evm/src/cpu/kernel/ast.rs:32]: nit: Maybe call it `MacroVar` to make it clear that it should only be used in macros, or add a comment.

**Review by wborgeaud — APPROVED** [review]
Looks good to me!

File: evm/src/cpu/kernel/assembler.rs (hunk 1 above)
Comment (dlubarov) [reply evm/src/cpu/kernel/assembler.rs:63]: `macro` seems to be reserved in Rust (i.e. not a keyword yet, but reserved for future use)

Comment (wborgeaud) [reply evm/src/cpu/kernel/assembler.rs:63]: Ah I see, I've usually seen people put the underscore after the name in these cases. You can also escape keywords using `r#macro`, although it's pretty ugly. Anyways, definitely not important.

**Pull request merged**